src/fidesops/ops/api/deps.py

```diff
@@ -1,26 +1,48 @@
 from typing import Generator
 
-from fideslib.db.session import get_db_session
+from fideslib.db.session import get_db_engine, get_db_session
+from sqlalchemy.orm import Session
 
 from fidesops.ops.common_exceptions import FunctionalityNotConfigured
 from fidesops.ops.core.config import config
 from fidesops.ops.util.cache import get_cache as get_redis_connection
 
+_engine = None
+
 
 def get_db() -> Generator:
     """Return our database session"""
     if not config.database.enabled:
         raise FunctionalityNotConfigured(
             "Application database required, but it is currently disabled! Please update your application configuration to enable integration with an application database."
         )
+
+    try:
+        db = _get_session()
+        yield db
+    finally:
+        db.close()
+
+
+def get_db_for_health_check() -> Generator:
+    """Gets a database session regardless of whether the application db is disabled, for a health check."""
     try:
-        SessionLocal = get_db_session(config)
-        db = SessionLocal()
+        db = _get_session()
         yield db
     finally:
         db.close()
 
 
+def _get_session() -> Session:
+    """Gets a database session"""
+    global _engine  # pylint: disable=W0603
```

Comment:
Good solution. Is there a limit to how many connections one engine can make?

Reply:
yes, I noted this in the description, we're currently using the defaults, so I believe the `pool_size` is 5, and there's a default `max_overflow` of 15.  This is something that we'll probably want to make configurable at some point.